single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -901,3 +893,32 @@ def _initialise_replica_set(self):
     def is_removing_last_replica(self) -> bool:
         """Returns True if the last replica (juju unit) is getting removed."""
         return self.state.planned_units == 0 and len(self.state.peers_units) == 0
+
+    def get_statuses(self) -> list[StatusBase]:
+        """Returns the statuses of the charm manager.."""
+        charm_statuses: list[StatusBase] = []
+
+        if not self.workload.workload_present:
+            charm_statuses.append(CharmStatuses.MONGODB_NOT_INSTALLED)
+        else:  # don't bother checking if started if not installed
+            if not self.state.db_initialised:
+                charm_statuses.append(CharmStatuses.mongodb.MONGODB_NOT_STARTED)
+
+            if not self.mongodb_exporter_config_manager.workload.active():
+                charm_statuses.append(CharmStatuses.mongodb.EXPORTER_NOT_STARTED)
+
+        if not self.state.is_sharding_component and self.state.has_sharding_integration:
+            charm_statuses.append(CharmStatuses.mongodb.SHARDING_ON_REPLICA)
+        elif (  # don't bother checking revision mismatch on sharding interface if replica
+            revision_mismatch_status
+            := self.cluster_version_checker.get_cluster_mismatched_revision_status()
+        ):
+            charm_statuses.append(revision_mismatch_status)
+
+        if not self.cluster_manager.is_valid_mongos_integration():
+            charm_statuses.append(CharmStatuses.mongodb.UNSUPPORTED_MONGOS_REL)
+
+        if not self.backup_manager.is_valid_s3_integration():
+            charm_statuses.append(CharmStatuses.mongodb.INVALID_S3_INTEGRATION_STATUS)
```

Comment:
For all this, you have to add the value of the enum not the enum entry itself.
While for many cases it's the same thing, it doesn't hold the same type (and mypy complains about it, see [here](https://github.com/canonical/mongo-single-kernel-library/actions/runs/13809027912/job/38626126234?pr=19#step:6:51)